pkg/controller/machine/controller.go

```diff
@@ -422,21 +422,23 @@ func isInvalidMachineConfigurationError(err error) bool {
 func (r *ReconcileMachine) setPhase(machine *machinev1.Machine, phase string, errorMessage string) error {
 	if stringPointerDeref(machine.Status.Phase) != phase {
 		klog.V(3).Infof("%v: going into phase %q", machine.GetName(), phase)
+		// Patch annotations before updating status
```

Comment:
Need more comments explaining the side effect.

Reply:
@michaelgugino I've exapanded on this and added a bit of an explanation of the side effect, does the comment make sense?